packages/contracts/src/AutoDriveCreditsReceiver.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {Pausable} from "@openzeppelin/contracts/utils/Pausable.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+
+contract AutoDriveCreditsReceiver is Ownable2Step, ReentrancyGuard, Pausable {
+    constructor() Ownable(msg.sender) {}
```

Comment:
The `constructor` is typically after storage items. Also, it would make sense to set the treasury address on contract instantiation.